### PR TITLE
updated browser-warning user agent parsing to precompiled regex

### PR DESCRIPTION
**Pull request by itsRaCl**
## Proposed Changes

- Fixes #8403
- Updated the BrowserError component 
- updated package.json with new script (`supported-browsers`)
- also added  `supportedBrowsers.ts` file

### CODE REVIEW

**Comment by itsRaCl** [comment]
@AdityaJ2305 can you reivew this and merge if you find the changes okay

File: package.json
Comment (AdityaJ2305) [inline package.json:1]: How are we getting config of browserslist particularly  production key array  from package.json as there is no any .browserslistrc  file ?

**Comment by AdityaJ2305** [comment]
Hey, we can add a script in build key which is something like this 
` "build": "npm run generate-build-meta && npm run supported-browsers && npm run build:react",
`
so that whenever there is any change in config of browserlist on the time of building updated regex will be generated ? 
let me know your views

**Comment by itsRaCl** [comment]
We can but I think we should ask one of the core devs before making this change..
This is changing the actual build process so better to ask before hand

File: package.json
Comment (itsRaCl) [reply package.json:1]: it's fetching it from the browserslist key in the `package.json` and depending on the `NODE_ENV` variable set during building the app it'll fetch the appropriate lists to refer to

**Comment by AdityaJ2305** [comment]
yes absolutely , but it is important to consider so that we don't need to run the command manually on configuration changes to generate updated regex . Let me ask to them

File: package.json
Comment (AdityaJ2305) [reply package.json:1]: got it thanks

**Pull request merged**